ipatests/test_integration/test_adtrust_install.py

```diff
@@ -303,3 +304,567 @@ def test_ipa_user_s4u2self_pac(self):
+            'dns_lookup_kdc = true', 'dns_lookup_kdc = false'
+        )
+        self.master.put_file_contents(paths.KRB5_CONF, new_krb5_cfg)
+        result = self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password,
+             "-U"], raiseonerr=False
+        )
+        assert error_msg not in result.stderr_text
+
+    def test_unattended_adtrust_install_uses_default_netbios_name(self):
+        """
+        ipa-adtrust-install unattended install should use default
+        netbios name rather than prompting for it.
+        https://fedorahosted.org/freeipa/ticket/3497
+        """
+        msg = (
+            'Enter the NetBIOS name for the IPA domain'
+            'Only up to 15 uppercase ASCII letters and '
+            'digits are allowed.'
+        )
+        result = self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password,
+             "-U"], raiseonerr=False
+        )
+        assert msg not in result.stdout_text
+
+    def test_adtrust_install_with_def_rid_base_values(self):
+        """
+        Test that ipa-adtrust-install install is successful
+        with default rid and secondary values
+        """
+        rid_base = '1000'
+        sec_rid_base = '100000000'
+        self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password,
+             "--rid-base", rid_base,
+             "--secondary-rid-base", sec_rid_base,
+             "-U"], raiseonerr=False
+        )
+
+    def test_ipa_adtrust_install_with_add_agents_option(self):
+        """
+        This testcase checks that ipa-adtrust-install
+        with --add-agents works without any error
+        on a non-replica IPA server
+        """
+        self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password,
+             "--add-agents",
+             "-U"], raiseonerr=False
+        )
+
+    def test_ipa_adtrust_install_with_add_sids_option(self):
+        """
+        This testcase checks that ipa-adtrust-install
+        with --add-sids option works without any error
+        """
+        msg = (
+            'adding SIDs to existing users and groups\n'
+            'This step may take considerable amount of time, please wait..'
+        )
+        result = self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password,
+             "--add-sids",
+             "-U"], raiseonerr=False
+        )
+        assert msg in result.stdout_text
+
+    def test_cldap_responder_doesnot_hang_for_domain_discovery(self):
+        """
+        This testcase checks that cldap responder doesnot hang
+        for domain discovery.
+        https://pagure.io/freeipa/issue/3639
+        """
+        base_dn = str(self.master.domain.basedn)
+        base = "(&(DnsDomain='{}')(NtVer=\06\00\00\00)" \
```

Comment:
The variable name is misleading: this is a filter, not a base

Reply:
@flo-renaud variable name has been changed to srch_filter from base.